### DISCUSSION TITLE
This error comes up sometime --> Invalid user object. It must be an instance of the "User" model

### DISCUSSION THREAD

**Discussion by salman3001** [adonisjs/core#4537] [General]
hi, 

i just want to discuss regarding this error. "Invalid user object. It must be an instance of the "User" model". it randomly comes up during development when trying to login user.  cant reporduce it because it happens only some time. i just restart dev server and it goes away. 

![image](https://github.com/adonisjs/core/assets/61388655/ad03ed1b-570b-412c-b425-b66a31b92ccb)

this is my auth service 

```
@inject()
export default class AuthService {
  constructor(protected ctx: HttpContext) {}
  async login() {
    const { email, password } = this.ctx.request.only(['email', 'password'])
    return await User.verifyCredentials(email, password)
  }
}
```

and this is my controller
```
  async login({ response, request, session, auth }: HttpContext) {
    const user = await this.authService.login()
    const next = request.qs()?.next

    if (user instanceof User) {
      await auth.use('web').login(user)
      session.flash('flash', {
        message: 'Login Success!',
        type: 'success',
      })
....
```

if anything wrong in my code ?, or it might be a bug ?

**Comment by GabrielNBDS** [comment]
I had that and updating @adonisjs/auth to latest fixed

**Reply by GabrielNBDS** [reply]
nope, i thought it fixed but it just stopped for some time

**Reply by GabrielNBDS** [reply]
i updated to node 21 and looks like it is fine now, i'll tell if it was not it

**Comment by thetutlage** [comment]
The issue comes in HMR mode and we can reproduce it. Will come back with more information soon.

**Comment by Julien-R44** [comment]
That was indeed a problem with Hot Hook. Please updgrade to `hot-hook@0.2.2` and the problem should be fixed. Let me know if it's ok for you

**Reply by salman3001** [reply]
Ok.. I ll try upgrading it.  

But if this error is only with hmr. Then it's fine. 
It appears only some times. And I don't mind restarting the server even.
